src/kokkos/test/kokkos/testEigenGPUNoFit.cc

```diff
@@ -0,0 +1,289 @@
+    Kokkos::View<Matrix3d, KokkosExecSpace> vm,
+    Kokkos::View<Eigen::SelfAdjointEigenSolver<Matrix3d>::RealVectorType, KokkosExecSpace> vret) {
+  eigenValues(vm.data(), vret.data());
+}
+
+KOKKOS_INLINE_FUNCTION void kernelInverse3x3(Kokkos::View<Matrix3d, KokkosExecSpace> vm,
+                                             Kokkos::View<Matrix3d, KokkosExecSpace> vmret) {
+  vmret() = vm().inverse();
+}
+
+KOKKOS_INLINE_FUNCTION void kernelInverse4x4(Kokkos::View<Matrix4d, KokkosExecSpace> vm,
+                                             Kokkos::View<Matrix4d, KokkosExecSpace> vmret) {
+  vmret() = vm().inverse();
+}
+
+KOKKOS_INLINE_FUNCTION void kernelInverse5x5(Kokkos::View<Matrix5d, KokkosExecSpace> vm,
+                                             Kokkos::View<Matrix5d, KokkosExecSpace> vmret) {
+  vmret() = vm().inverse();
+}
+
+template <typename M1, typename M2, typename M3>
+KOKKOS_INLINE_FUNCTION void kernelMultiply(Kokkos::View<M1, KokkosExecSpace> d_j,
+                                           Kokkos::View<M2, KokkosExecSpace> d_c,
+                                           Kokkos::View<M3, KokkosExecSpace> d_result) {
+//  Map<M3> res(result->data());
+#if TEST_DEBUG
+  printf("*** GPU IN ***\n");
+#endif
+  printIt(d_j.data());
+  printIt(d_c.data());
+  //  res.noalias() = (*J) * (*C);
+  //  printIt(&res);
+  d_result() = d_j() * d_c();
+#if TEST_DEBUG
+  printf("*** GPU OUT ***\n");
+#endif
+  return;
+}
+
+template <int row1, int col1, int row2, int col2>
+void testMultiply() {
+  std::cout << "TEST MULTIPLY" << std::endl;
+  std::cout << "Product of type " << row1 << "x" << col1 << " * " << row2 << "x" << col2 << std::endl;
+
+  Kokkos::View<Matrix<double, row1, col1>, KokkosExecSpace> d_j("d_j");
+  Kokkos::View<Matrix<double, row2, col2>, KokkosExecSpace> d_c("d_c");
+  Kokkos::View<Matrix<double, row1, col2>, KokkosExecSpace> d_multiply_result("d_multiply_result");
+
+  auto h_j = Kokkos::create_mirror_view(d_j);
+  auto h_c = Kokkos::create_mirror_view(d_c);
+  auto h_multiply_result = Kokkos::create_mirror_view(d_multiply_result);
+
+  fillMatrix(h_j());
+  fillMatrix(h_c());
+  h_multiply_result() = h_j() * h_c();
+  auto multiply_result = h_multiply_result();
+
+#if TEST_DEBUG
+  std::cout << "Input J:" << std::endl;
+  printIt(h_j.data());
+  std::cout << "Input C:" << std::endl;
+  printIt(h_c.data());
+  std::cout << "Output:" << std::endl;
+  printIt(&multiply_result);
+#endif
+  // GPU
+  Eigen::Matrix<double, row1, col2>* multiply_resultGPUret = new Eigen::Matrix<double, row1, col2>();
+
+  Kokkos::deep_copy(KokkosExecSpace(), d_j, h_j);
+  Kokkos::deep_copy(KokkosExecSpace(), d_c, h_c);
+  Kokkos::deep_copy(KokkosExecSpace(), d_multiply_result, h_multiply_result);
+
+  team_policy policy = team_policy(KokkosExecSpace(), 1, 1);
+  Kokkos::parallel_for(
+      "kernelMultiply", policy, KOKKOS_LAMBDA(const member_type& teamMember) {
+        kernelMultiply(d_j, d_c, d_multiply_result);
```

Comment:
In principle this one could use `RangePolicy`, no?

Reply:
Sure, I can update these functions too. I didn't do it since they are all 1x1 kernels.

Reply:
Thanks. Although it's not really a big deal, I view `RangePolicy` simpler than `TeamPolicy` and would prefer to use the simplest hammer :)

Reply:
Done